app/kubemci/cmd/create.go

```diff
@@ -161,34 +163,54 @@ func runCreate(options *CreateOptions, args []string) error {
 
 // Extracts the contexts from the given kubeconfig and creates ingress in those context clusters.
 // Returns the list of clusters in which it created the ingress and a map of clients for each of those clusters.
-func createIngress(kubeconfig string, kubeContexts []string, ing *v1beta1.Ingress) ([]string, map[string]kubeclient.Interface, error) {
+func createIngress(kubeconfig string, kubeContexts []string, ing *v1beta1.Ingress, forceUpdate bool) ([]string, map[string]kubeclient.Interface, error) {
 	clients, err := kubeutils.GetClients(kubeconfig, kubeContexts)
 	if err != nil {
 		return nil, nil, err
 	}
-	clusters, createErr := createIngressInClusters(ing, clients)
+	clusters, createErr := createIngressInClusters(ing, clients, forceUpdate)
 	return clusters, clients, createErr
 }
 
 // Creates the given ingress in all clusters corresponding to the given clients.
-func createIngressInClusters(ing *v1beta1.Ingress, clients map[string]kubeclient.Interface) ([]string, error) {
+func createIngressInClusters(ing *v1beta1.Ingress, clients map[string]kubeclient.Interface, forceUpdate bool) ([]string, error) {
 	var err error
 	var clusters []string
 	for cluster, client := range clients {
 		glog.V(4).Infof("Creating Ingress in cluster: %v...", cluster)
 		clusters = append(clusters, cluster)
 		glog.V(3).Infof("Using namespace %s for ingress %s", ing.Namespace, ing.Name)
-		actualIng, createErr := client.Extensions().Ingresses(ing.Namespace).Create(ing)
-		glog.V(2).Infof("Ingress Create returned: err:%v. Actual Ingress:%+v", createErr, actualIng)
-		if createErr != nil {
-			if errors.IsAlreadyExists(createErr) {
-				fmt.Println("Ingress already exists; moving on.")
-				continue
+		existingIng, getErr := client.ExtensionsV1beta1().Ingresses(ing.Namespace).Get(ing.Name, metav1.GetOptions{})
+		if getErr != nil {
+			if errors.IsNotFound(getErr) {
+				actualIng, createErr := client.ExtensionsV1beta1().Ingresses(ing.Namespace).Create(ing)
+				glog.V(2).Infof("Ingress Create returned: err:%v. Actual Ingress:%+v\n", createErr, actualIng)
+				if createErr != nil {
+					err = multierror.Append(err, fmt.Errorf("Error in creating ingress in cluster %s: %s", cluster, createErr))
+				}
 			} else {
-				err = multierror.Append(err, fmt.Errorf("Error in creating ingress in cluster %s: %s", cluster, createErr))
+				err = multierror.Append(err, fmt.Errorf("Error in checking for existing ingress in cluster %s: %s", cluster, getErr))
+			}
+			fmt.Println("Created Ingress for cluster:", cluster)
```

Comment:
nit: would recommend changing this to "Created Ingress *in* cluster" to go with the "Creating Ingress *in* cluster" message before.